src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -104,9 +113,25 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     throw;
                 }
+                finally
+                {
+                    telemetryClients?.TrackEvent(telemetryEventDetails, TelemetryConstants.AcquireTokenEventName);
+                }
             }
         }
 
+        private void LogSuccessfulTelemetry(AuthenticationResult authenticationResult, MsalTelemetryEventDetails telemetryEventDetails)
```

Comment:
>LogSuccessfulTelemetry [](http://example.com/codeflow?start=21&length=22)

take a look at how this is done in SAL, to my eye it's much more readable and clear what's going on. Perhaps just a style thing (S2STelemetryEventDetails).